### PR TITLE
PIM-3890: add flat denormalizer

### PR DESCRIPTION
**Pull request by skeleton**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | Y
| BC breaks?           | N
| CI currently passed? | ?
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-3890
| Doc PR               | N
| Related CE PR        | N

### CODE REVIEW

File: spec/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizerSpec.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    {
+        $this->denormalize('', 'pim_reference_data_multiselect', 'csv')
+            ->shouldBeAnInstanceOf('Doctrine\Common\Collections\ArrayCollection');
+        $this->denormalize(null, 'pim_reference_data_multiselect', 'csv')
+            ->shouldBeAnInstanceOf('Doctrine\Common\Collections\ArrayCollection');
+        $this->denormalize([], 'pim_reference_data_multiselect', 'csv')
+            ->shouldBeAnInstanceOf('Doctrine\Common\Collections\ArrayCollection');
+    }
+
+    function it_throws_an_exception_if_context_value_is_not_an_product_value_inteface()
```
Comment (grena) [inline spec/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizerSpec.php]: `not_a_product`

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizer.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public function denormalize($data, $referenceDataClass, $format = null, array $context = array())
+    {
+        $collection = new ArrayCollection();
+        if (empty($data)) {
+            return $collection;
+        }
+
+        if (!$context['value'] instanceof ProductValueInterface) {
+            throw new InvalidParameterException(
+                sprintf('Value is not an instance of Pim\Bundle\CatalogBundle\Model\ProductValueInterface.')
```
Comment (jjanvier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizer.php]: no need sprint apparently

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizer.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            );
+        }
+
+        $attribute = $context['value']->getAttribute();
+        if (null === $attribute) {
+            throw new InvalidParameterException(
+                sprintf('Denormalizer\'s context expected to have an attribute, none found.')
```
Comment (jjanvier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataCollectionDenormalizer.php]: same

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataDenormalizer.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function denormalize($data, $referenceDataClass, $format = null, array $context = array())
+    {
+        if (empty($data)) {
+            return null;
+        }
+
+        if (!$context['value'] instanceof ProductValueInterface) {
+            throw new InvalidParameterException(
+                sprintf('Value is not an instance of Pim\Bundle\CatalogBundle\Model\ProductValueInterface.')
```
Comment (jjanvier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataDenormalizer.php]: same

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataDenormalizer.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            );
+        }
+
+        $attribute = $context['value']->getAttribute();
+        if (null === $attribute) {
+            throw new InvalidParameterException(
+                sprintf('Denormalizer\'s context expected to have an attribute, none found.')
```
Comment (jjanvier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/ReferenceDataDenormalizer.php]: same

**Pull request merged**